File: program/program/calculate-the-permutation-of-n-objects-taken-r-at-a-time/calculate_the_permutation_of_n_objects_taken_r_at_a_time.py
def npr(n, r):
    if r > n:
        return 0
    # n! / (n-r)!  = n*(n-1)*(n-2)*..*(n-r+1)*(n-r)! / (n-r)! =n*(n-1)*(n-2)*..*(n-r+1)
    result = 1
    for i in range(n, n - r, -1):
        result *= i
    return result


n = int(input("Enter the n value: "))
r = int(input("Enter the r value: "))
print("npr value is:", npr(n, r))

File: program/program/calculate-the-permutation-of-n-objects-taken-r-at-a-time/CalculateThePermutationOfNObjectsTakenRAtATime.swift
import Foundation

// Function to calculate factorial
func factorial(_ n: Int) -> Int {
    return (n == 0) ? 1 : n * factorial(n - 1)
}

// Function to calculate permutations
func permutation(_ n: Int, _ r: Int) -> Int {
    return factorial(n) / factorial(n - r)
}

// Taking input from the user
print("input:", terminator: " ")
let inputs = readLine()?.split(separator: ",").compactMap { Int($0.trimmingCharacters(in: .whitespaces)) }
if let n = inputs?[0], let r = inputs?[1] {
    // Calculating permutations
    let result = permutation(n, r)

    // Displaying the result
    print(result)
} else {
    print("Invalid input")
}
